src/main/filesystem.ts

```diff
@@ -75,14 +75,16 @@ export class LauncherFileSystem {
   }
 
   happUiDir(appId: string, holochainDataRoot: HolochainDataRoot) {
-    switch (holochainDataRoot.type) {
-      case 'partition':
-        return path.join(this.holochainPartitionDir(holochainDataRoot.name), 'apps', appId, 'ui');
-      case 'external':
-        return path.join(holochainDataRoot.path, 'apps', 'ui');
-      default:
-        throw new Error('Got invalid data root type: ', (holochainDataRoot as any).type);
+    const holochainDataRootParsed = HolochainDataRootSchema.safeParse(holochainDataRoot);
```

Comment:
I think we should not do any parsing here because this function `happUiDir` is called each time an asset is being served to the happ's browser window. So I think it's worth keeping this method as cheaply as possible, i.e. not do any strictly necessary computation to not cause performance bottlenecks and slowly loading UIs.